src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -118,19 +171,30 @@ private ObservedDataMapping copyObservedData(ObservedDataMapping observedDataMap
          };
       }
 
+      private static bool isLocalFileAndExists(string file)

```

Comment:
`isLocalFileAndExists` sounds strange for me. 
Why not just `fileExists `or `localFileExists`?

Reply:
well this returns true if the file is a local file and exists. 
FileExists would be wrong because the file might be remote and does not exist local
localFileExists or isLocalFileAndExists is semantically equivalent. Not sure what to say here

Reply:
let's go with `localFileExists `then

Reply:
No problem